x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_setup_state.tsx

```diff
@@ -3,35 +3,82 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
-import { useState, useCallback } from 'react';
 
-type SetupHandler = (startTime?: number | undefined, endTime?: number | undefined) => void;
+import { useState, useCallback, useMemo } from 'react';
 
-interface Props {
+import { isExampleDataIndex } from '../../../../common/log_analysis';
+
+type SetupHandler = (
+  indices: string[],
+  startTime: number | undefined,
+  endTime: number | undefined
+) => void;
+
+interface AnalysisSetupStateArguments {
+  availableIndices: string[];
   cleanupAndSetupModule: SetupHandler;
   setupModule: SetupHandler;
 }
 
+type IndicesSelection = Record<string, boolean>;
+
+type ValidationErrors = 'TOO_FEW_SELECTED_INDICES';
+
 const fourWeeksInMs = 86400000 * 7 * 4;
 
-export const useAnalysisSetupState = ({ setupModule, cleanupAndSetupModule }: Props) => {
+export const useAnalysisSetupState = ({
+  availableIndices,
+  cleanupAndSetupModule,
+  setupModule,
+}: AnalysisSetupStateArguments) => {
   const [startTime, setStartTime] = useState<number | undefined>(Date.now() - fourWeeksInMs);
   const [endTime, setEndTime] = useState<number | undefined>(undefined);
 
+  const [selectedIndices, setSelectedIndices] = useState<IndicesSelection>(
+    availableIndices.reduce(
+      (indexMap, indexName) => ({
+        ...indexMap,
+        [indexName]: !(availableIndices.length > 1 && isExampleDataIndex(indexName)),
+      }),
+      {}
+    )
+  );
+
+  const selectedIndexNames = useMemo(
+    () =>
+      Object.entries(selectedIndices)
+        .filter(([_indexName, isSelected]) => isSelected)
+        .map(([indexName]) => indexName),
+    [selectedIndices]
+  );
+
   const setup = useCallback(() => {
-    return setupModule(startTime, endTime);
-  }, [setupModule, startTime, endTime]);
+    return setupModule(selectedIndexNames, startTime, endTime);
+  }, [setupModule, selectedIndexNames, startTime, endTime]);
 
   const cleanupAndSetup = useCallback(() => {
-    return cleanupAndSetupModule(startTime, endTime);
-  }, [cleanupAndSetupModule, startTime, endTime]);
+    return cleanupAndSetupModule(selectedIndexNames, startTime, endTime);
+  }, [cleanupAndSetupModule, selectedIndexNames, startTime, endTime]);
+
+  const validationErrors: ValidationErrors[] = useMemo(
+    () => [
```

Comment:
If I understood this function correctly, I think it can be simplified a bit:

```ts
() => Object.values(selectedIndices).some(isSelected => isSelected)
        ? []
        : ['TOO_FEW_SELECTED_INDICES' as const]
```

Reply:
True, I was thinking of other possible validation criteria :wink: I'll simplify it until we have more.